stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -2679,8 +2677,9 @@ internal constructor(
                 appSettingsManager = appSettingsManager,
                 chatSocketExperimental = module.experimentalSocket(),
                 lifecycle = lifecycle
-            ).also {
-                configureInitializer(it)
+            ).also { chatClient ->
+                chatClient.clientState = ClientState.create(appContext)
```

Comment:
I wonder if we can move the `clientState` property into the `ChatClient` constructor, considering that `clientState` is getting initialized right after `ChatClient` creation.

Reply:
So we could create the `ClientState` instance in `BaseChatModule` and provide it to `ChatClient`.